packages/ripple/index.js

```diff
@@ -0,0 +1,174 @@
+
+import {MDCRippleFoundation, util} from '@material/ripple';
+
+const withRipple = (WrappedComponent) => {
+  class RippledComponent extends Component {
+
+    foundation_ = null;
+    adapter_ = null;
+    instance_ = null;
+
+    state = {
+      classList: new Set(),
+      style: {},
+    };
+
+    componentDidMount() {
+      if (!this.adapter_) {
+        throw new Error('You must call initRipple from the element\'s ' +
+          'ref prop to initialize the adapter for withRipple');
+      }
+      this.foundation_ = new MDCRippleFoundation(this.adapter_);
+      this.foundation_.init();
+    }
+
+    componentWillUnmount() {
+      if (this.foundation_) {
+        this.foundation_.destroy();
+      }
+    }
+
+    createAdapter = (instance) => {
+      const adapter = {
+        browserSupportsCssVars: () => util.supportsCssVariables(window),
+        isUnbounded: () => this.props.unbounded,
+        isSurfaceActive: () => this.state.isActivated,
+        isSurfaceDisabled: () => this.props.disabled,
+        addClass: (className) =>
+          this.setState({classList: this.state.classList.add(className)}),
+        removeClass: (className) => {
+          const {classList} = this.state;
+          classList.delete(className);
+          this.setState({classList});
+        },
+        registerDocumentInteractionHandler: (evtType, handler) =>
+          document.documentElement.addEventListener(evtType, handler, util.applyPassive()),
+        deregisterDocumentInteractionHandler: (evtType, handler) =>
+          document.documentElement.removeEventListener(evtType, handler, util.applyPassive()),
+        registerResizeHandler: (handler) => window.addEventListener('resize', handler),
+        deregisterResizeHandler: (handler) => window.removeEventListener('resize', handler),
+        updateCssVariable: this.updateCssVariable,
+        computeBoundingRect: () => instance.getBoundingClientRect(),
+        getWindowPageOffset: () => ({x: window.pageXOffset, y: window.pageYOffset}),
+      };
+
+      this.instance_ = instance;
+      this.adapter_ = adapter;
+    }
+
+    get classes() {
+      const {className: wrappedCompClasses} = this.props;
+      const {classList} = this.state;
+
+      return classnames(Array.from(classList), wrappedCompClasses);
+    }
+
+    handleEvent = (e, eventType) => {
+      const {onMouseDown, onMouseUp, onTouchStart, onKeyDown} = this.props;
+
+      let action;
+      switch (eventType) {
+        case 'onMouseDown':
+          action = onMouseDown;
+          this.setState({isActivated: true});
+          break;
+        case 'onMouseUp':
+          action = onMouseUp;
+          this.setState({isActivated: false});
+          break;
+        case 'onTouchStart':
+          action = onTouchStart;
+          break;
+        case 'onKeyDown':
+          action = onKeyDown;
+          break;
+      }
+
+      if (eventType !== 'onMouseUp') {
+        this.activateRipple(e);
+      }
+      action(e);
```

Comment:
instead of saving the function in another variable (action) and then calling it...just call the original function. Much simpler.